### PR TITLE
constraint pages

### PR DESCRIPTION
**Pull request by sploiselle**
Converted `constraints.md` as a branch page to individual pages for each constraint:

- `check.md`
- `default-value.md`
- `foreign-key.md`
- `not-null.md`
- `primary-key.md`
- `unique.md`

Also updated all links and changed sidebar structure to surface Constraints as third-level items.

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/901)

### CODE REVIEW

File: constraints.md [hunk 1]
```diff
@@ -6,287 +6,22 @@ toc: false
 
 Constraints offer additional data integrity by enforcing conditions on the data within a column or row. They are checked during DML operations and restrict the data values within a column to those specified within the constraint.
 
-If a constraint refers to only one column (column-level constraint) it can be defined against the column as part of its definition. If a constraint refers to more than one column (table-level constraint) it needs to be defined as a separate entry in the tables definition.  
+If a constraint refers to only one column (column-level constraint) it can be defined against the column as part of its definition. If a constraint refers to more than one column (table-level constraint) it needs to be defined as a separate entry in the tables definition.
```
Comment (jseldess) [inline constraints.md]: nit: In both sentences, add a comma after the closing parenthesis to separate the clauses.

File: check.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+The Check [constraints](constraints.html) specifies that the column value must satisfy a Boolean expression.
+
+<div id="toc"></div>
+
+## Details
+
+- Check constraints requires that the column values satisfy a Boolean expression within the constraint. The expression must evaluate to `TRUE` (or *NULL*) for every row affected by an `INSERT` or `UPDATE` statement. The DML statement will fail if the condition evaluates to `FALSE` for any row.
+- Check constraints may be specified at the column or table level and can reference other columns within the table. Internally, all column level Check constrints are converted to table level constraints so they can be handled in a consistent fashion.
+- You can have multiple Check constraints on a single column but ideally these should be combined using the logical operators. So, for example,
```
Comment (sploiselle) [inline check.md]: Yeah––it's running two independent checks and combining the results vs. a single check. Just computational efficiency.

Comment (jseldess) [inline check.md]: OK, that makes sense. Whenever a "should"-type recommendation is used, as a reader, I prefer a little of the reasoning. How about:

> ... but ideally, for computational efficiency, they should be combined using the logical operators.

File: check.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  warranty_period INT CHECK (warranty_period >= 0) CHECK (warranty_period <= 24)
+  ~~~
+  
+  should be specified as 
+  
+  ~~~ sql
+  warranty_period INT CHECK (warranty_period BETWEEN 0 AND 24)
+  ~~~
+
+- Check constraints that refer to multiple columns should be specified at the [table level](#multiple-column-table-level).
```
Comment (jseldess) [inline check.md]: Right, sorry to loop that edit in here. We could handle that in a separate PR, if you like.

File: check.md (hunk 2 above)
Comment (dt) [inline check.md]: it'll be slightly cheaper to evaluate fewer expressions, and we can potentially optimize things away, etc so yeah, when it doesn't make it harder to understand/maintain, folding multiple CHECK expressions into one is good, but the savings aren't usually going to be a big deal.

File: check.md (hunk 3 above)
Comment (dt) [inline check.md]: All `CHECK` constraints get moved to the table level anyway, so it doesn't really matter.

**Pull request merged**